hudi-common/src/main/java/org/apache/hudi/common/table/read/HoodieFileGroupReader.java

```diff
@@ -334,6 +334,8 @@ public HoodieReadStats getStats() {
    */
   T next() {
     T nextVal = recordBuffer == null ? baseFileIterator.next() : recordBuffer.next();
+    Schema requiredSchema = readerContext.getSchemaHandler().getRequiredSchema();
+    nextVal = readerContext.fromBinaryRow(requiredSchema, nextVal);
```

Comment:
why we pass around the required schema here as long as the reader context already holds it.